File: LeetCode/3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.py
class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        st = set()
        back = -k
        n = len(nums)
        
        for i in range(n):
            temp = max(back + 1, nums[i] - k)
            if temp <= nums[i] + k:
                st.add(temp)
                back = temp
        
        return len(st)

File: LeetCode/3397-maximum-number-of-distinct-elements-after-operations/3397-maximum-number-of-distinct-elements-after-operations.java
class Solution {
    public int maxDistinctElements(int[] nums, int k) {
        Arrays.sort(nums);
        Set<Integer> st = new HashSet<>();
        int back = -k;
        int n = nums.length;
        
        for (int i = 0; i < n; i++) {
            int temp = Math.max(back + 1, nums[i] - k);
            if (temp <= nums[i] + k) {
                st.add(temp);
                back = temp;
            }
        }
        
        return st.size();
    }
}